src/code/z_actor.c

```diff
@@ -222,186 +212,180 @@ void Actor_SetFeetPos(Actor* actor, s32 limbIndex, s32 leftFootIndex, Vec3f* lef
     }
 }
 
-void func_8002BE04(GlobalContext* globalCtx, Vec3f* arg1, Vec3f* arg2, f32* arg3) {
-    SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->mf_11D60, arg1, arg2, arg3);
-    *arg3 = (*arg3 < 1.0f) ? 1.0f : (1.0f / *arg3);
+void Math_GetProjectionPos(GlobalContext* globalCtx, Vec3f* src, Vec3f* xyzDest, f32* wDest) {
+    SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->projectionMatrix, src, xyzDest, wDest);
+    *wDest = (*wDest < 1.0f) ? 1.0f : (1.0f / *wDest);
 }
 
-typedef struct {
-    /* 0x00 */ Color_RGBA8 inner;
-    /* 0x04 */ Color_RGBA8 outer;
-} NaviColor; // size = 0x8
-
-NaviColor sNaviColorList[] = {
-    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },
-    { { 255, 255, 255, 255 }, { 0, 0, 255, 0 } },     { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },
-    { { 150, 150, 255, 255 }, { 150, 150, 255, 0 } }, { { 255, 255, 0, 255 }, { 200, 155, 0, 0 } },
-    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },
-    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         { { 255, 255, 0, 255 }, { 200, 155, 0, 0 } },
-    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },
-    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },
-};
+void Target_SetPos(TargetContext* targetCtx, s32 targetIndex, f32 x, f32 y, f32 z) {
+    targetCtx->targetInfo[targetIndex].pos.x = x;
+    targetCtx->targetInfo[targetIndex].pos.y = y;
+    targetCtx->targetInfo[targetIndex].pos.z = z;
+    targetCtx->targetInfo[targetIndex].radius = targetCtx->targetRadius;
+}
 
-// unused
-Gfx D_80115FF0[] = {
-    gsSPEndDisplayList(),
+// Used for both the Z Target and Navi when targetting
+static Color_RGBA8 sNaviTargetColorList[][2] = {
```

Comment:
We try to avoid naming stuff in the codebase based on character names (for example Player vs Link). So we should avoid using `Navi` too.
```suggestion
static Color_RGBA8 sFairyColorList[][2] = {
```

Reply:
Calling it fairy is a slight issue because other fairies technically exist in the game (i.e. the ones belonging to the Kokiri children)

Reply:
Renaming this to anything besides Navi is outside the scope of this pr I think since it's called Navi everywhere else in the repo, so I think I'll have to resolve this.